jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcDataContext.java

```diff
@@ -778,18 +780,33 @@ private String findDefaultSchema(final String defaultName, final List<String> sc
      * @throws SQLException
      */
     private Set<String> getSchemaSQLServerNames(DatabaseMetaData metaData) throws SQLException {
-        // Distinct schema names. metaData.getTables() is a denormalized
-        // resultset
-        Set<String> schemas = new HashSet<>();
-        ResultSet rs = null;
+        // Distinct schema names. metaData.getTables() is a denormalized resultset
+        final Set<String> schemas = new HashSet<>();
+        ResultSet resultSet = null;
+
+        // Find DBO schema if present
         try {
-            rs = metaData.getTables(_catalogName, null, null, JdbcUtils.getTableTypesAsStrings(_tableTypes));
-            while (rs.next()) {
-                schemas.add(rs.getString("TABLE_SCHEM"));
+            resultSet = metaData.getSchemas();
+            while (resultSet.next()) {
+                if (resultSet.getString("TABLE_SCHEM").equals(DEFAULT_SCHEMA_NAME_SQLSERVER)) {
+                    schemas.add(DEFAULT_SCHEMA_NAME_SQLSERVER);
+                    break;
+                }
             }
         } finally {
-            FileHelper.safeClose(rs);
+            FileHelper.safeClose(resultSet);
         }
+
+        // Find other schemas
+        try {
```

Comment:
Same here as on line 788, since `ResultSet` is `AutoCloseable`, isn't it better to change this to try-with-resources?

Reply:
See previous answer.

Reply:
Implemented try-with-resources